tornado/test/process_test.py

```diff
@@ -195,6 +197,8 @@ def test_stderr(self):
         subproc.stderr.read_until(b'\n', self.stop)
         data = self.wait()
         self.assertEqual(data, b'hello\n')
+        # More mysterious EBADF: This fails if done with self.addCleanup instead of here.
```

Comment:
Perhaps because stdout and stderr share the same file descriptor (you used `stderr=subprocess.STDOUT` above)?